### ISSUE TITLE
Greenshot has various issues on a high DPI screen

### ISSUE THREAD

**Issue by RMerl** [greenshot/greenshot#540]
This GitHub repository is for developers, if you want to report a bug for Greenshot as a user please do so in our JIRA issue system here: https://greenshot.atlassian.net/projects/BUGS/issues/filter=allopenissues

**Describe the bug**
There are a number of things within Greenshot that seem to have scaling issues on a high DPI screen.  For instance when drawing an arrow, the arrow head is minuscule compared to a regular DPI screen.  Or if you try to add text on a capture, the text will me impossibly small.

**To Reproduce**
Example steps to reproduce the behavior:
1. Have the monitor set to a high DPI (like 150%-200%)
2. Capture something, and edit it in the Editor
3. Try to add an arrow, or to add text - both will be unusually small

**Expected behavior**
Text and arrow should be proportional to the display

**Screenshots**
![image](https://github.com/greenshot/greenshot/assets/1661562/332aa96d-b6b7-41e6-b453-29e653e3b09f)

**Versions (please complete the following information):**
 - Greenshot version: 1.3.277 (cannot read the tag following it, DPI issues on the about window makes other text overlap it...)
 - Windows version: Windows 11 23H2

**Comment by tdp4** [comment]
I have this issue too.  Scaling is not implemented.  The side icons are incredibly small too.
![image](https://github.com/user-attachments/assets/fc471078-540c-44f6-baaa-4dc29614b7b2)

**Comment by bworline** [comment]
Agreed.  And even if you manually adjust the icon size, the dialog text size can be small.

Here is the settings dialog on a 3840x2160 screen:

<img width="295" alt="image" src="https://github.com/user-attachments/assets/31fb8174-cec8-4010-a208-aaa11a36cbff">

And here is the same dialog when I am remoted into a machine running Greenshot on the remote (same screen & dimensions):

<img width="343" alt="image" src="https://github.com/user-attachments/assets/4cbc11b9-2bfe-46ca-893c-c96f868b9fc5">

**Comment by daxliniere** [comment]
Yes, I too can confirm that the scaling is incorrect (side icons too tiny to see). I tried Windows' High DPI Scaling Override for greenshot.exe, and it fixes the small icons, but makes the whole program unusable as when you press PrtScrn to take a screenshot, the entire display becomes low resolution (think Duplo vs Lego) and so you can't capture the screen region you wanted.

**Comment by darkstar** [comment]
I just installed Greenshot and it looks like this :-D
Windows 11, DPI Scaling 100%

![Image](https://github.com/user-attachments/assets/d330c3df-9b46-4a0d-a021-ace5a08cc5d5)

**Comment by hmax-72** [comment]
Similar issue here. Icons and menu in the editor are incredibly small on a 4k screen. I'm on the latest version.
As I'm working with different setups (work/home) changing the settings all the time is not an option.

**Comment by Lakritzator** [comment]
@hmax-72 I really do not need you to comment every single ticket with the same unusable information. Greenshot is provided free, thus limiting the time I can invest in it. Unfortunately the tech we used, Windows forms, is very limited when it comes to good high DPI support. There are ways to have an easy solution, as some people use compatibility settings, this can also be done by us, BUT Greenshot must be DPI aware to do it's work correctly as it is a low level tool which needs to know how big Windows on the screen are without being lied to. So the setting makes the UI look better, but breaks our interactions with the OS. This puts tools like Greenshot, which use Windows forms (WinForms) in a bad corner.

We are working on trying to fix this (see #1156), but people already gave up in other tries (one fix made other parts unusable) and we are working to move towards newer UI tech (see #784), but to make this possible we also need to revise other parts (huge change in #1149). In parallel we are working on other topics too, e.g. keep Greenshot stable and secure. So I hope you understand this takes time. 

Time I just "wasted" by answering this and being pinged due to you commenting everywhere, that was more time than I had this morning. Also I want to make sure you understand: I create Greenshot for free, in my **limited** spare time, which I cannot plan, because of personal reasons. I do it to enjoy myself working on a challenge, or solve issues I have, not to be criticized by people. This is one reason I didn't work on it for 2 years, I need motivation not complaints. If people do what you do, some are way worse, I rather work on other parts and ignore that noise, I even block access to the repository. I realize that maybe you didn't think about my side, but it's really annoying and doesn't add ANYTHING.

Another hard fact: I do not have a 4K display, for me personally the issue is not that urgent. Do not translate that into we won't fix it, as I said we are working on it, just maybe not today. Can you do something to speed this up? Yes, stop spamming the project and deliver 2x a ROG PG27UCDM on my doorstep 😄😛
